lib/ansible/modules/network/mlnxos/mlnxos_command.py

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/python
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'network'}
+
+
+DOCUMENTATION = """
+---
+module: mlnxos_command
+extends_documentation_fragment: mlnxos
+version_added: "2.5"
+author: "Samer Deeb (@samerd)"
+short_description: Run commands on remote devices running MLNXOS
+description:
+  - >-
+       Sends arbitrary commands to an mlnxos node and returns the results
+       read from the device. This module includes an
+       argument that will cause the module to wait for a specific condition
+       before returning or timing out if the condition is not met.
+  - >-
+      This module does not support running commands in configuration mode.
+      Please use M(mlnxos_config) to configure MLNXOS devices.
+notes:
+  - tested on Mellanox OS 3.6.4000
+options:
+  commands:
+    description:
+      - >-
+           List of commands to send to the remote mlnxos device over the
+           configured provider. The resulting output from the command
+           is returned. If the I(wait_for) argument is provided, the
+           module is not returned until the condition is satisfied or
+           the number of retries has expired.
+    required: true
+  wait_for:
+    description:
+      - >-
+          List of conditions to evaluate against the output of the
+          command. The task will wait for each condition to be true
+          before moving forward. If the conditional is not true
+          within the configured number of retries, the task fails.
+          See examples.
+    required: false
+    default: null
+    aliases: ['waitfor']
```

Comment:
Please remove this line.
`waitfor` was used in older modules, we've since standardized on `wait_for`